### PR TITLE
Moving WSC span realignment to loading time

### PR DESCRIPTION
**Pull request by pruksmhc**
This PR currently holds: 
-Moving the span realignment to loading time as opposed to a separate preprocessing step that needs to be run before running the experiment.
-Fixing the WSC class so that it matches the format of the final version of the data 
-Some fixes to the WSC configuration so that it matches the experiments

Left to do:
-Point the test for preprocessing to the new location of realign_spans
-A test for loading checkpoints

### CODE REVIEW

File: config/superglue-bert.conf [hunk 1]
```diff
@@ -8,7 +8,6 @@ exp_name = "bert-large-cased"
 max_seq_len = 256 // Mainly needed for MultiRC, to avoid over-truncating
                   // But not 512 as that is really hard to fit in memory.
 tokenizer = "bert-large-cased"
-
```
Comment (sleepinyourhat) [inline config/superglue-bert.conf]: Restore?

File: config/superglue-bert.conf [hunk 2]
```diff
@@ -42,3 +41,7 @@ do_full_eval = 1
 write_preds = "val,test"
 write_strict_glue_format = 1
 
+// For WSC
```
Comment (sleepinyourhat) [inline config/superglue-bert.conf]: Why is this in this PR?

File: src/tasks/tasks.py [hunk 3]
```diff
@@ -2092,11 +2081,11 @@ def _stream_records(self, filename):
     def load_data(self):
         iters_by_split = collections.OrderedDict()
         for split, filename in self._files_by_split.items():
-            iter = list(self._stream_records(filename))
-            iters_by_split[split] = iter
+            if "test" in filename:
```
Comment (sleepinyourhat) [inline src/tasks/tasks.py]: This seems brittle. What if someone tries to evaluate on a span task with a public test set?

File: src/trainer.py [hunk 4]
```diff
@@ -1229,6 +1229,7 @@ def _restore_checkpoint(self, search_phases_in_priority_order=["pretrain"]):
 
         metric_states = torch.load(metric_state_path)
         for metric_name, metric_state in metric_states.items():
+
```
Comment (sleepinyourhat) [inline src/trainer.py]: Cut?

File: src/utils/data_loaders.py [hunk 5]
```diff
@@ -11,11 +11,40 @@
...
+    Returns:
+        List of dictionaries of the aligned spans and tokenized text.
+    """
+    rows = pd.read_json(file_name, lines=True)
+    # realign spans
+    rows = rows.apply(lambda x: realign_spans(x, tokenizer_name), axis=1)
+
+    if has_labels is False:
+        rows["label"] = False
+    return list(rows.T.to_dict().values())
```
Comment (sleepinyourhat) [inline src/utils/data_loaders.py]: Cut?

File: src/utils/data_loaders.py [hunk 6]
```diff
@@ -11,11 +11,40 @@
...
+        target: dict that contains the spans  
+    Args:
+        tokenizer_name: str, 
+        file_name: str, 
+        label_fn: function that expects a row and outputs a transformed row with labels tarnsformed.
+    Returns:
+        List of dictionaries of the aligned spans and tokenized text.
+    """
+    rows = pd.read_json(file_name, lines=True)
+    # realign spans
```
Comment (sleepinyourhat) [inline src/utils/data_loaders.py]: Explain what this does, either here or in the header comment.

File: config/superglue-bert.conf (hunk 1 above)
Comment (pruksmhc) [reply config/superglue-bert.conf]: Not sure why it's appearing in github as removed because it isn't

File: src/tasks/tasks.py (hunk 3 above)
Comment (sleepinyourhat) [reply src/tasks/tasks.py]: This still seems dangerously brittle...

File: src/utils/data_loaders.py [hunk 7]
```diff
@@ -11,11 +11,40 @@
...
 
+def load_span_data(
+        tokenizer_name, 
+        file_name,
+        label_fn=None,
+        has_labels=True
+    ):
+    """
+    Load a span-related task file in .jsonl format, does 
+    re-alignment of spans, and tokenizes the text.
```
Comment (sleepinyourhat) [inline src/utils/data_loaders.py]: Do we explain what this means anywhere?

**Pull request merged**